splunk/src/main/java/com/splunk/Receiver.java

```diff
@@ -94,26 +93,19 @@ public Socket attach(String indexName, Args args) throws IOException {
             postUrl = postUrl + args.encode();
         }
 
-        StringBuilder header = new StringBuilder(String.format(
-                "%s HTTP/1.1\r\n" +
-                "Host: %s:%d\r\n" +
-                "Accept-Encoding: identity\r\n" +
-                "X-Splunk-Input-Mode: Streaming\r\n",
-                postUrl,
-                service.getHost(), service.getPort()
-                ));
+        List<String> headers = new ArrayList<>();
+        headers.add(String.format("%s HTTP/1.1", postUrl));
+        headers.add("Accept-Encoding: identity");
+        headers.add("X-Splunk-Input-Mode: Streaming");
 
         if (service.hasCookies()) {
-            header.append("Cookie: ");
-            header.append(service.stringifyCookies());
+            headers.add(String.format("Cookie: %s", service.stringifyCookies()));
         } else {
-            header.append("Authorization: ");
-            header.append(service.getToken());
+            headers.add(String.format("Authorization: %s", service.getToken()));
         }
-        header.append("\r\n\r\n");
-
-        out.write(header.toString());
-        out.flush();
+        headers.add("");
```

Comment:
Curious what this line is doing

Reply:
This empty string for headers is intentionally kept for terminating header string